metadata/org.testcontainers/testcontainers/1.19.8/reflect-config.json

```diff
@@ -0,0 +1,4219 @@
+[
```

Comment:
This is pretty huge `reflect-config.json`. Are you sure that all of these entries are necessary? For example there are lots of entries like this:
```
{
        "name": "java.util.LinkedHashMap",
        "queryAllDeclaredMethods": true,
        "condition": {
            "typeReachable": "org.testcontainers.dockerclient.DockerClientProviderStrategy"
        },
        "queryAllDeclaredConstructors": true
    },
    {
        "name": "java.util.LinkedHashMap",
        "condition": {
            "typeReachable": "org.testcontainers.dockerclient.EnvironmentAndSystemPropertyClientProviderStrategy"
        }
    },
    {
        "name": "java.util.LinkedHashMap",
        "condition": {
            "typeReachable": "org.testcontainers.dockerclient.TestcontainersHostPropertyClientProviderStrategy"
        }
    },
    {
        "name": "java.util.LinkedHashMap",
        "queryAllDeclaredMethods": true,
        "condition": {
            "typeReachable": "org.testcontainers.shaded.com.fasterxml.jackson.databind.ObjectMapper"
        },
        "queryAllDeclaredConstructors": true
    },
```

or something like 
```
  {
        "name": "java.net.UnixDomainSocketAddress",
        "condition": {
            "typeReachable": "org.testcontainers.dockerclient.DockerClientProviderStrategy"
        },
        "methods": [
            {
                "name": "of",
                "parameterTypes": [
                    "java.lang.String"
                ]
            }
        ]
    },
```

do we really need to include all of this stuff? Please take a quick look whether there are entries that can be excluded or not? If you find such entries, please update your [user-code-filter.json](https://github.com/oracle/graalvm-reachability-metadata/pull/533/files#diff-16ce117efd6cc1e3d7b2a4e2d8adb3e30d6af5628baede0fb46a81fdc3d84132R2) to exclude them

Reply:
While I do not know the specifics of the type entries, when we compare what's currently in the repository for testcontainers (1.17.6), and diff with the new config, we get 8 new types and 1 type removed. Existing classes get many, many more conditions though, explaining the number of entries going from 358 to 452. So while the file is big, it's not adding many types to the resulting image. Details below.

---

Here are the 8 new types:

```
com.github.dockerjava.api.model.LoadResponseItem
com.github.dockerjava.api.model.RuntimeInfo
java.util.concurrent.ForkJoinTask
java.util.concurrent.atomic.AtomicMarkableReference
org.testcontainers.shaded.com.github.dockerjava.core.DockerContextMetaFile
org.testcontainers.shaded.com.github.dockerjava.core.DockerContextMetaFile$Endpoints
org.testcontainers.shaded.com.github.dockerjava.core.DockerContextMetaFile$Endpoints$Docker
sun.security.provider.SHA2$SHA256
```

And the 1 type removed, `[Lorg.testcontainers.shaded.com.google.common.collect.ImmutableMapEntry;` . 


We do have many more conditions per type, though. For example, `com.github.dockerjava.api.model.AuthConfig` went from 1.17.6 :

```json
{
  "condition": {
    "typeReachable": "org.testcontainers.DockerClientFactory"
  },
  "name": "com.github.dockerjava.api.model.AuthConfig",
  "allDeclaredFields": true,
  "allDeclaredConstructors": true
}
```

to 1.19.7:

```json
{
  "condition": {
    "typeReachable": "org.testcontainers.DockerClientFactory"
  },
  "name": "com.github.dockerjava.api.model.AuthConfig",
  "allDeclaredFields": true,
  "allDeclaredConstructors": true
}
{
  "name": "com.github.dockerjava.api.model.AuthConfig",
  "condition": {
    "typeReachable": "org.testcontainers.containers.GenericContainer"
  }
}
{
  "name": "com.github.dockerjava.api.model.AuthConfig",
  "condition": {
    "typeReachable": "org.testcontainers.containers.GenericContainer$$Lambda$478/0x000000e80123bd50"
  }
}
{
  "name": "com.github.dockerjava.api.model.AuthConfig",
  "queryAllDeclaredMethods": true,
  "condition": {
    "typeReachable": "org.testcontainers.dockerclient.DockerClientProviderStrategy"
  },
  "queryAllDeclaredConstructors": true
}
{
  "name": "com.github.dockerjava.api.model.AuthConfig",
  "condition": {
    "typeReachable": "org.testcontainers.dockerclient.EnvironmentAndSystemPropertyClientProviderStrategy"
  }
}
{
  "name": "com.github.dockerjava.api.model.AuthConfig",
  "condition": {
    "typeReachable": "org.testcontainers.dockerclient.TestcontainersHostPropertyClientProviderStrategy"
  }
}
{
  "name": "com.github.dockerjava.api.model.AuthConfig",
  "condition": {
    "typeReachable": "org.testcontainers.shaded.com.fasterxml.jackson.databind.ObjectMapper"
  },
  "allDeclaredFields": true,
  "methods": [
    {
      "name": "getAuth",
      "parameterTypes": []
    },
    {
      "name": "getEmail",
      "parameterTypes": []
    },
    {
      "name": "getIdentitytoken",
      "parameterTypes": []
    },
    {
      "name": "getPassword",
      "parameterTypes": []
    },
    {
      "name": "getRegistryAddress",
      "parameterTypes": []
    },
    {
      "name": "getRegistrytoken",
      "parameterTypes": []
    },
    {
      "name": "getStackOrchestrator",
      "parameterTypes": []
    },
    {
      "name": "getUsername",
      "parameterTypes": []
    }
  ]
}
{
  "name": "com.github.dockerjava.api.model.AuthConfig",
  "condition": {
    "typeReachable": "org.testcontainers.shaded.com.github.dockerjava.core.DefaultDockerClientConfig$Builder"
  }
}
{
  "name": "com.github.dockerjava.api.model.AuthConfig",
  "condition": {
    "typeReachable": "org.testcontainers.shaded.com.github.dockerjava.core.DockerConfigFile"
  },
  "allDeclaredFields": true,
  "methods": [
    {
      "name": "<init>",
      "parameterTypes": []
    }
  ]
}
{
  "name": "com.github.dockerjava.api.model.AuthConfig",
  "condition": {
    "typeReachable": "org.testcontainers.shaded.com.github.dockerjava.core.command.CreateContainerCmdImpl"
  }
}
{
  "name": "com.github.dockerjava.api.model.AuthConfig",
  "condition": {
    "typeReachable": "org.testcontainers.shaded.com.github.dockerjava.core.exec.AbstrDockerCmdExec"
  }
}
{
  "name": "com.github.dockerjava.api.model.AuthConfig",
  "condition": {
    "typeReachable": "org.testcontainers.shaded.com.github.dockerjava.core.exec.CreateContainerCmdExec"
  }
```

I don't see myself combing through all the new configs manually - and I'm not familiar with Testcontainers internals to begin with.